WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -392,12 +392,35 @@ def time_until_competition
     }
   end
 
-  def currency_convert
-    update_rates_if_needed
-    converted = Money.new(params[:value], params[:from_currency]).exchange_to(params[:to_currency])
+  def calculate_dues
+    country_iso2 = Country.where(id: params[:country_id]).pluck(:iso2)[0]
+    country_band = CountryBand.where(iso2: country_iso2).map(&:number)[0]
+
+    update_currency_rates_if_needed
+    currency_subunit = Money::Currency.new(params[:currency_code]).subunit_to_unit
+    money_value = params[:entry_fee].to_i / currency_subunit
+    money_value_usd = Money.new(money_value, params[:currency_code]).exchange_to("USD").cents
+
+    base_fee_usd = money_value_usd * 0.15
+    band_fee_usd = 0
+    if country_band.present? && country_band > 0
+      band_fee_usd = CountryBand::BANDS[country_band][:value]
+    end
+
+    # times 100 because later currency conversions require lowest currency subunit, which is cents for USD
```

Comment:
General remark about money variables (didn't want to bother marking the entire method body): Please pick variable names that make clear whether you're currently handling a `Money` object-oriented model or an integer representing cents.

For example. the below calculation result should be called something like `price_per_competitor_us_cents`. Similar remarks apply for other variables throughout this method body.